Content.Server/Radiation/Systems/RadiationSystem.Blockers.cs

```diff
@@ -0,0 +1,171 @@
+using Content.Shared.Doors;
+using Content.Shared.Doors.Components;
+using Content.Shared.Radiation.Components;
+
+namespace Content.Server.Radiation.Systems;
+
+// create and update map of radiation blockers
+public partial class RadiationSystem
+{
+    // dict of grid uid -> grid pos -> resistance value
+    // no float value - no resistance on this tile
+    // if grid has no resistance tile - it should be deleted
+    private readonly Dictionary<EntityUid, Dictionary<Vector2i, float>> _resistancePerTile = new();
```

Comment:
Don't store data on systems.

I'm also skeptical if this is worth it over iterating anchored entities with an entityquery.

Reply:
About second part. I tested anchored entities approach against current implementation.

Using radiation grid:
```
[INFO] root: Radiation update: 0.2625ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.3488ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2868ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2689ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2798ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2977ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2596ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.2727ms with. Receivers: 21, Sources: 1, Rays: 21
```
Using get anchored entities:
```
[INFO] root: Radiation update: 0.8021ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.8242ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.9302ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.8116ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.8272ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.8423ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.825ms with. Receivers: 21, Sources: 1, Rays: 21
[INFO] root: Radiation update: 0.848ms with. Receivers: 21, Sources: 1, Rays: 21
```

Reply:
Here the test code
```csharp
var radsResistance = 0f;
if (GridcastUseRadGrid)
{
    if (resistanceMap.TryGetValue(point, out var resData))
        radsResistance = resData;
}
else
{
    var ents = grid.Grid.GetAnchoredEntities(point);
    foreach (var uid in ents)
    {
        if (!blocker.TryGetComponent(uid, out var blc))
            continue;
        radsResistance += blc.RadResistance;
    }
}
```

Reply:
So imo the impact is sufficient enough. I think the main reason is because grid traversal need to visit same tiles over and over again, while caching them in map is done only on change.

Reply:
I moved radiation resistance data to grid component. I also added a new visualization of rad blockers into radiation debug overlay.

<img width="412" alt="Снимок экрана 2022-09-03 в 00 34 19" src="https://user-images.githubusercontent.com/6161335/188242978-17912e8b-3550-437e-ba7d-6b7a0d7eec76.png">

Reply:
> Don't store data on systems.

While not really relevant to this PR anymore given that its apparently on a grid component now, this should still be resolved given that explosions also do that.

I think this came up in a discord discussion which I can't find anymore, but I think the idea is that its fine to store data related to processing a system on the system, but not entity/state data (& obviously extra care has to be taken for shared/client systems for prediction resetting).

In this case the radiation (/explosion) blocking map is sort of a weird case where its a tool to assist in radiation system processing, and is not really part of single entity state given that it depends on the combination of many anchored entities, so I think its probably fine to have on a system? Though making it a grid component automatically removes some boilerplate related to clearing dictionary entities on grid deletion, which is nice,